pysat/_orbits.py

```diff
@@ -817,18 +816,9 @@ def __iter__(self):
         while self.sat.empty:
             self.sat.next()
 
-        # if self.sat._iter_type == 'file':
-        #     for fname in self.sat._iter_list:
-        #         self.sat.load(fname=fname)
-        #         break
-        #
-        # elif self.sat._iter_type == 'date':
-        #     for date in self.sat._iter_list:
-        #         self.sat.load(date=date)
-        #         break
-        # else:
-        #     raise ValueError('Iteration type not set')
-
         while True:
-            self.next()
-            yield self.sat
+            try:
+                self.next()
```

Comment:
How did this exit previously?

Reply:
Previously, the `StopIteration` raised in the `.next()` would do exactly that.  With python 3.7, `StopIteration` raises a `RuntimeError` and crashes.  The new syntax is what other instances of the pysat iterators use, I think this is just something that slipped through the cracks.  We didn't see it until 3.7 forced us to.